cfn-resources/tool/markdown-generator/generate_version_markdown.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+)
+
+const mongoDBPublisherID = "bb989456c78c398a858fef18f2ca1bfc1fbba082"
+const ouputMarkdownFile = "resource-versions.md"
+
+func main() {
+	// can potentially get regions from github publish yaml file
+	var regions = []string{"af-south-1", "ap-east-1", "ap-northeast-1", "ap-northeast-2", "ap-northeast-3", "ap-south-1", "ap-southeast-1", "ap-southeast-2", "ap-southeast-3",
+		"ca-central-1", "eu-central-1", "eu-north-1", "eu-south-1", "eu-west-1", "eu-west-2", "eu-west-3", "me-south-1", "sa-east-1", "us-east-1", "us-east-2", "us-west-1", "us-west-2",
+		"ap-south-2", "ap-southeast-4", "eu-central-2", "eu-south-2", "me-central-1"}
```

Comment:
Should this be defined as ENV var? I am wondering if we should pass the regions as input to make it easier once the script run as part of an Action

Reply:
+1 in not hardcoding it.

My main point in one previous comment was: do we have to specify the regions at all? can we take them from a common public library (like the aws-sdk-go) or are we forced in selecting a subset of regions?

Reply:
Thanks for clarifying.
We do currently support a subset of regions (the ones which are supported by Atlas https://www.mongodb.com/docs/atlas/reference/amazon-aws/#amazon-web-services--aws-).
With this in mind, I will adjust to receive regions as env variable for this execution. When defining the github action to periodically trigger this generation, we can look into referencing the default regions values used for publishing (https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/.github/workflows/cfn-publish-resource.yaml#L8) so we have a single source for our supported regions.